packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -516,6 +516,8 @@ export class TreeCheckout implements ITreeCheckoutFork {
 					event.type === "append" ||
 					(event.type === "replace" && getChangeReplaceType(event) === "transactionCommit")
 				) {
+					// TODO:#20949: When the SharedTree is detached, these commits will already have been garbage collected.
```

Comment:
Note that the _current_ (before this PR) behavior is that all reverts will do nothing when the SharedTree is detached.

The _new_ (after this PR) behavior is that _some_ reverts (those requiring no refreshers) will succeed and others (those requiring refreshers) will throw an error.

This is because the `change` property of the commit that has been garbage collected is no longer overwritten with an empty changeset.

A followup PR will fix the underlying bug so that all reverts will succeed.